packit_service/worker/helpers/testing_farm.py

```diff
@@ -354,40 +358,61 @@ def distro2compose(self, distro: str, arch: str) -> str:
+            logger.error(msg)
+            self.report_status_to_test_for_test_target(
+                state=BaseCommitStatus.error,
+                description=msg,
+                target=target,
+            )
+            return None
+
+        # {'composes': [{'name': 'CentOS-Stream-8'}, {'name': 'Fedora-Rawhide'}]}
+        composes = [c["name"] for c in response.json()["composes"]]
+
         if self.job_config.use_internal_tf:
-            # Internal TF does not have own endpoint for composes
-            # This should be solved on the TF side.
-            # For more information on the composes
-            # see https://api.dev.testing-farm.io/v0.1/composes/redhat
+            if compose in composes:
+                return compose
+
             if compose == "Fedora-Rawhide":
-                return "Fedora-Rawhide-Nightly"
-            if compose.startswith("Fedora-"):
-                return f"{compose}-Updated"
-            if compose.startswith("CentOS") and len(compose) == len("CentOS-7"):
+                compose = "Fedora-Rawhide-Nightly"
+            elif compose.startswith("Fedora-"):
+                compose = f"{compose}-Updated"
+            elif compose.startswith("CentOS") and len(compose) == len("CentOS-7"):
                 # Attach latest suffix only to major versions:
                 # CentOS-7 -> CentOS-7-latest
                 # CentOS-8 -> CentOS-8-latest
                 # CentOS-8.4 -> CentOS-8.4
                 # CentOS-8-latest -> CentOS-8-latest
                 # CentOS-Stream-8 -> CentOS-Stream-8
-                return f"{compose}-latest"
-            if compose == "RHEL-6":
-                return "RHEL-6-LatestReleased"
-            if compose == "RHEL-7":
-                return "RHEL-7-LatestReleased"
-            if compose == "RHEL-8":
-                return "RHEL-8.5.0-Nightly"
-            if compose == "Oracle-Linux-7":
-                return "Oracle-Linux-7.9"
-            if compose == "Oracle-Linux-8":
-                return "Oracle-Linux-8.6"
-        else:
-            response = self.send_testing_farm_request(endpoint="composes")
-            if response.status_code == 200:
-                # {'composes': [{'name': 'CentOS-Stream-8'}, {'name': 'Fedora-Rawhide'}]}
-                composes = [c["name"] for c in response.json()["composes"]]
-                if compose not in composes:
-                    logger.error(f"Can't map {compose} (from {distro}) to {composes}")
+                compose = f"{compose}-latest"
+            elif compose == "RHEL-6":
+                compose = "RHEL-6-LatestReleased"
+            elif compose == "RHEL-7":
+                compose = "RHEL-7-LatestReleased"
+            elif compose == "RHEL-8":
+                compose = "RHEL-8.5.0-Nightly"
+            elif compose == "Oracle-Linux-7":
+                compose = "Oracle-Linux-7.9"
+            elif compose == "Oracle-Linux-8":
+                compose = "Oracle-Linux-8.6"
```

Comment:
I don't see Oracle-Linux-X.Y in https://api.dev.testing-farm.io/v0.1/composes/redhat so I'm wondering what's the status. @thrix ?